src/app/data-model-provider/BUILD.gn

```diff
@@ -24,6 +24,16 @@ source_set("metadata") {
   ]
 }
 
+source_set("core") {
+  public = [ "OperationTypes.h" ]
```

Comment:
why did we split?

Beside that, please do not name things core/support/helpers and other filler words.